### PR TITLE
Add 'document changed' indicator

### PR DESCRIPTION
**Pull request by tjsilver**
## What does this change?

Currently a user has no way of knowing if Typerighter should run again, given that the document has changed in a significant way. This could mislead a user into thinking a document has been checked when it hasn't.

This PR adds a small blue dot to indicate when a document has changed since the last time the 'Check document' button was pressed.

## How to test
Run the branch locally and make a change to the document. See the blue dot appear. Click 'Check document' and see it disappear until the document is changed again.

## How can we measure success?
Users are able to tell if the matches shown are as up-to-date as possible.

## Have we considered potential risks?
Users might not realise what the dot is for, or might find it irritating.

## Images
doc_changed

### CODE REVIEW

**Review by jonathonherbert — APPROVED** [review]
Code looks great. Tested locally, works as expected. Top work.

As @SHession has mentioned, if there's a precedent in other UIs for moving the button to the right (GH does this, for example) it might be worth following that convention – IIRC western eyes read left to right and alerts or actions are often found there as a result, but I don't have strong opinions.

It'd be great to see that title text when we hover over the button – the dot is very small, and I think people are less likely to see it as a result.

**Review by SHession — APPROVED** [review]
Code looks get and tested locally. 

+1 to @jonathonherbert's comments on title text and placement.

File: src/ts/components/Controls.tsx [hunk 1]
```diff
@@ -115,24 +116,54 @@ const Controls = <TPluginState extends IPluginState>({
     );
   };
 
+  const renderCheckDocumentButton = () => {
+    if (!pluginState) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (SHession) [reply src/ts/components/Controls.tsx:122]: This looks like a sensible change to me 👍

[the remark this answers is not in the document]
Comment (jonathonherbert) [reply src/ts/components/Controls.tsx:122]: Hm, if we haven't got any plugin state, good luck to the user clicking the check button! This is such an unusual state that it seems odd to defend against, but nonetheless for completeness I think this is a good idea.

**Comment by tjsilver** [comment]
[Typerighter indicating dirtied state](https://trello.com/c/dIiryQtt/449-typerighter-indicating-dirtied-state)

**Review by jonathonherbert — APPROVED** [review]

**Pull request merged**